### ISSUE TITLE
False positive for ModifiedButNotUsed check is reported

### ISSUE THREAD

**Issue by davido** [google/error-prone#1622]
The EP@HEAD (8beaab4aa02e62a1fb8a6f3f6493f39b6196ffa8) is erroneously reporting variable as unused:

```
java/com/google/gerrit/server/plugins/PluginGuiceEnvironment.java:348: error: [ModifiedButNotUsed] A collection or proto builder was created, but its values were never accessed.
      Map<Annotation, ReloadableRegistrationHandle<?>> am = new HashMap<>();
```

The source is here: [1]. The `am` variable is accessed in line 366:

```java
ReloadableRegistrationHandle<Object> h =
            (ReloadableRegistrationHandle<Object>) am.remove(key.getAnnotation());
```

[1] https://github.com/GerritCodeReview/gerrit/blob/master/java/com/google/gerrit/server/plugins/PluginGuiceEnvironment.java#L348-L366

**Comment by cushon** [comment]
I'm pretty sure this was fixed in ccd34868e0704c6978cdb29cd069b1fb4fc5bca9.

I can reproduce against gerrit with the version of EP currently in Bazel, but not with EP at head. How did you determine it's still happening at head? Do you have a repro?

**Comment by davido** [comment]
> How did you determine it's still happening at head? Do you have a repro?

I bumped EP version twice in Bazel to  2.3.5-SNAPSHOT: [1]. And uploaded two changes to activate more checks in Gerrit. In the first change: [2] I activated new checks for old EP version, and in second change: [3] I bumped `java_tools` to my custom release: [4] that I conducted with EP 2.3.5-SNAPSHOT.

Unfortunately, I misread the build breakage. The breakage was reported for the first change, with the old EP version and not the second. Sorry for the noise.

[1] https://github.com/bazelbuild/bazel/pull/11426
[2] https://gerrit-review.googlesource.com/c/gerrit/+/267952
[3] https://gerrit-review.googlesource.com/c/gerrit/+/265493
[4] https://github.com/davido/java_tools/releases/tag/javac11-v12.0
